commands.go

```diff
@@ -1,329 +1,352 @@
-		},
-	})
-
-	if err != nil {
-		log.Print(err)
-	}
-}
-
-func leaveCTFCallback(s *discordgo.Session, i *discordgo.InteractionCreate, ctfName string) {
-	guild, err := s.Guild(i.GuildID)
-	if err != nil {
-		s.InteractionRespond(i.Interaction, &discordgo.InteractionResponse{
-			Type: discordgo.InteractionResponseChannelMessageWithSource,
-			Data: &discordgo.InteractionResponseData{
-				Flags:   1 << 6,
-				Content: "Couldn't find the calling Guild",
-			},
-		})
-		return
-	}
-
-	callingUser := i.Member.User
-	log.Printf("Removing user %s from CTF %s\n", callingUser.Username, ctfName)
-
-	// Find the role of the CTF and add the user to it
-	var ctfRole *discordgo.Role
-	for _, r := range guild.Roles {
-		if r.Name == ctfName {
-			ctfRole = r
-		}
-	}
-	// Check if the role was ever found and correct if not
-	if ctfRole == nil {
-		s.InteractionRespond(i.Interaction, &discordgo.InteractionResponse{
-			Type: discordgo.InteractionResponseChannelMessageWithSource,
-			Data: &discordgo.InteractionResponseData{
-				Flags:   1 << 6,
-				Content: "Role " + ctfName + " does not exist. Try creating it, first!",
-			},
-		})
-		return
-	}
-
-	s.GuildMemberRoleRemove(guild.ID, callingUser.ID, ctfRole.ID)
-
-	// Reply with success
-	err = s.InteractionRespond(i.Interaction, &discordgo.InteractionResponse{
-		Type: discordgo.InteractionResponseChannelMessageWithSource,
-		Data: &discordgo.InteractionResponseData{
-			Flags:   1 << 6,
-			Content: "Removed user " + callingUser.Username + " from role " + ctfRole.Name,
-		},
-	})
-
-	if err != nil {
-		log.Print(err)
-	}
-}
+package main
+
+import (
+	"log"
+
+	"github.com/bwmarrin/discordgo"
+)
+
+// ID for the CTF text channel category
+const ctfCategoryID = "1259233177994661910"
+
+// Define our ApplicationCommands
+var (
+	// Create an array of ApplicationCommand structs to register the definitions of our commands
+	commands = []*discordgo.ApplicationCommand{
+		{
+			Name:        "ping",
+			Description: "ping-command",
+		},
+		{
+			Name:        "ctf",
+			Type:        discordgo.ChatApplicationCommand,
+			Description: "Parent command for the CTF group",
+			// Subcommands for 'ctf'
+			Options: []*discordgo.ApplicationCommandOption{
+				{
+					Name:        "create",
+					Type:        discordgo.ApplicationCommandOptionSubCommand,
+					Description: "Create a CTF",
+					Options: []*discordgo.ApplicationCommandOption{
+						{
+							Type:        discordgo.ApplicationCommandOptionString,
+							Name:        "ctf-name",
+							Description: "CTF name",
+							Required:    true,
+						},
+					},
+				},
+				{
+					Name:        "join",
+					Type:        discordgo.ApplicationCommandOptionSubCommand,
+					Description: "Join a CTF",
+					Options: []*discordgo.ApplicationCommandOption{
+						{
+							Type:        discordgo.ApplicationCommandOptionString,
+							Name:        "ctf-name",
+							Description: "CTF name",
+							Required:    false,

```

Comment:
Should this be required?

Reply:
agree w/ @quantatic  here, probably make this required